owncloudData/src/test/java/com/owncloud/android/data/files/repository/OCFileRepositoryTest.kt

```diff
@@ -25,382 +26,2010 @@ import com.owncloud.android.data.files.datasources.LocalFileDataSource
+        localFileDataSource,
+        remoteFileDataSource,
+        localSpacesDataSource,
+        localStorageProvider
     )
+    private val ocFileRepositorySpy = spyk(ocFileRepository)
+
+    private val expectedRemotePath = OC_FOLDER_WITH_SPACE_ID.remotePath + OC_FILE_WITH_SPACE_ID.fileName
+    private val storagePath = "/local/storage/path/username@demo.owncloud.com/Folder/Photos/image2.jpt"
+    private val remoteId = "remoteId"
+    private val searchText = "image"
+    private val etagInConflict = "5efb0c13c688i"
+    private val fileWithConflict = OC_FILE_WITH_SPACE_ID.copy(etagInConflict = etagInConflict)
+
+    @Before
+    fun setUp() {
+        val commonSpaceId = OC_FILE_WITH_SPACE_ID.spaceId
+        val commonAccountName = OC_FILE_WITH_SPACE_ID.owner
+        every {
+            localSpacesDataSource.getWebDavUrlForSpace(
+                spaceId = commonSpaceId,
+                accountName = commonAccountName
+            )
+        } returns OC_SPACE_PERSONAL.root.webDavUrl
+    }
+
+    @Test
+    fun `createFolder creates a new folder and saves it`() {
+        every {
+            localSpacesDataSource.getWebDavUrlForSpace(
+                spaceId = OC_PARENT_FOLDER_WITH_SPACE_ID.spaceId,
+                accountName = OC_PARENT_FOLDER_WITH_SPACE_ID.owner
+            )
+        } returns OC_SPACE_PERSONAL.root.webDavUrl
+        // The result of this method is not used, so it can be anything
+        every { localFileDataSource.saveFilesInFolderAndReturnTheFilesThatChanged(any(), any()) } returns emptyList()
 
-    private val timeInMilliseconds = 3600000L
+        ocFileRepository.createFolder(OC_FOLDER_WITH_SPACE_ID.remotePath, OC_PARENT_FOLDER_WITH_SPACE_ID)
+
+        verify(exactly = 1) {
+            localSpacesDataSource.getWebDavUrlForSpace(
+                spaceId = OC_PARENT_FOLDER_WITH_SPACE_ID.spaceId,
+                accountName = OC_PARENT_FOLDER_WITH_SPACE_ID.owner
+            )
+            remoteFileDataSource.createFolder(
+                remotePath = OC_FOLDER_WITH_SPACE_ID.remotePath,
+                createFullPath = false,
+                isChunksFolder = false,
+                accountName = OC_PARENT_FOLDER_WITH_SPACE_ID.owner,
+                spaceWebDavUrl = OC_SPACE_PERSONAL.root.webDavUrl
+            )
+            localFileDataSource.saveFilesInFolderAndReturnTheFilesThatChanged(
+                folder = OC_PARENT_FOLDER_WITH_SPACE_ID,
+                listOfFiles = withArg<List<OCFile>> {
+                    assertEquals(1, it.size)
+                    assertEquals(OC_FOLDER_WITH_SPACE_ID.remotePath, it[0].remotePath)
+                    assertEquals(OC_PARENT_FOLDER_WITH_SPACE_ID.owner, it[0].owner)
+                    assertEquals(0, it[0].length)
+                    assertEquals(MIME_DIR, it[0].mimeType)
+                    assertEquals(OC_PARENT_FOLDER_WITH_SPACE_ID.spaceId, it[0].spaceId)
+                    assertEquals("CK", it[0].permissions)
+                }
```

Comment:
Don't you include a check to verify if there are conflicts when creating a folder?

Reply:
That check is in another class, so that will be tested in the corresponding tests. In this case, in the repository we just stick to the code we have, that is calling to remote datasource and local datasource afterwards, that's why it's done like this

Reply:
👍